src/SFML/Audio/SoundFileReaderMp3.cpp

```diff
@@ -0,0 +1,130 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2021 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#define MINIMP3_IMPLEMENTATION // Minimp3 control define, places implementation in this file.
+#include <SFML/Audio/SoundFileReaderMp3.hpp>
+#include <SFML/System/MemoryInputStream.hpp>
+#include <SFML/System/Err.hpp>
+#include <algorithm>
+#include <cstring>
+
+
+namespace
+{
+std::size_t readCallback(void* ptr, std::size_t size, void* data)
+{
+    sf::InputStream* stream = static_cast<sf::InputStream*>(data);
+    return static_cast<std::size_t>(stream->read(ptr, size));
+}
+
+int seekCallback(std::uint64_t offset, void* data)
+{
+    sf::InputStream* stream = static_cast<sf::InputStream*>(data);
+    sf::Int64 position = stream->seek(offset);
+    return position < 0 ? -1 : 0;
+}
+}
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+bool SoundFileReaderMp3::check(InputStream& stream)
+{
+    Uint8 header[10];
+    if (stream.read(header, sizeof(header)) < sizeof(header))
+        return false;
+    if (!memcmp(header, "ID3", 3) && !((header[5] & 15) || (header[6] & 0x80) || (header[7] & 0x80) || (header[8] & 0x80) || (header[9] & 0x80)))
```

Comment:
Maybe short explanation regarding the header format would help understanding here.
The second part, the `&& !(...)` checks if _none_ of the bytes 5 to 9 match a certain value/bit pattern?

Reply:
@lieff do you have some insights to what this checks exactly?

Reply:
Hi) Mp3 detection usually more complex and ideally mp3dec_detect_cb should be used here. But big portion of file can be read here to do it properly.
So I write here light version that only checks for ID3 tag and first valid frame header. The `&& !(...)` part checks that ID3 tag header is valid.
This light method will fail on l3-sin1k0db.bit test vector for example, and can give false positives on non-mp3 files (because only 1 mp3 frame header is checked). But should work on most normal mp3 files.

Reply:
Ah okay, I'll change the check to `std::memcmp(...) == 0` as it's IMHO more understandable as the negated version and add a short comment. 
I think for now, this is good enough. If people run into issues, then we'll have to double check again. 👍